core/src/main/java/org/apache/carbondata/core/datastore/impl/FileFactory.java

```diff
@@ -43,11 +45,19 @@
    */
   private static final LogService LOGGER =
       LogServiceFactory.getLogService(FileFactory.class.getName());
-  private static Configuration configuration = null;
 
-  static {
-    configuration = new Configuration();
-    configuration.addResource(new Path("../core-default.xml"));
+  public static Configuration getConfiguration() {
+    Configuration configuration;
+    if (ThreadLocalSessionInfo.getCarbonSessionInfo() == null) {
+      configuration = new Configuration();
+      configuration.addResource(new Path("../core-default.xml"));
+    } else {
+      CarbonConfiguration carbonConfiguration =
+          (CarbonConfiguration) ThreadLocalSessionInfo.getCarbonSessionInfo().getThreadParams()
```

Comment:
Also handle case carbonConf not present even ThreadLocalSessionInfo is present, required for other than spark cases